### PR TITLE
`@functor Returns`

### PR DESCRIPTION
**Pull request by mcabbott**
Recently wanted this, and it seems natural given Fix1 etc.

### CODE REVIEW

**Review by ToucheSir — COMMENTED** [review]
Needs either a `Compat` dep or a version check to make 1.6 CI happy.

File: src/base.jl [hunk 1]
```diff
@@ -10,6 +10,8 @@
 @functor Base.Fix2
 @functor Base.Broadcast.BroadcastFunction
 
+@functor Base.Returns
```
Comment (ToucheSir) [inline src/base.jl:13]:
```suggestion
@static if VERSION >= v"1.7"
  @functor Base.Returns
end
```

Comment (mcabbott) [reply src/base.jl:13]: Thanks! I got distracted, think this is cleaner than adding Compat.jl

File: test/base.jl [hunk 2]
```diff
@@ -50,6 +50,12 @@ end
   @test fmap(x -> x + 10, bf) == Base.Broadcast.BroadcastFunction(Bar(13.3))
 end
 
+@testset "Returns" begin
+  ret = Returns([0, pi, 2pi])
+  @test Functors.functor(ret)[1] == (value = [0, pi, 2pi],)
+  @test Functors.functor(ret)[2]((value = 1:3,)) === Returns(1:3)
+end
```
Comment (ToucheSir) [inline test/base.jl:57]: Forgot this, I think it needs a VERSION guard too. If that's enough to pass, merge when ready.

**Review by ToucheSir — APPROVED** [review]

File: test/base.jl (hunk 2 above)
Comment (mcabbott) [reply test/base.jl:57]:
```suggestion
VERSION >= v"1.7" && @testset "Returns" begin
  ret = Returns([0, pi, 2pi])
  @test Functors.functor(ret)[1] == (value = [0, pi, 2pi],)
  @test Functors.functor(ret)[2]((value = 1:3,)) === Returns(1:3)
end
```

**Pull request merged**